### PR TITLE
Enhance Device Control Response and Add AnyIoCtlCode

### PR DESCRIPTION
**Pull request by ibeckermayer**
- Introduced `AnyIoCtlCode` to handle cases with unknown or non-critical IoCtlCode.
- Updated `DeviceControlResponse` to accept an optional `output_buffer`, aligning with FreeRDP's implementation for empty buffer scenarios.
- Implemented decoding for `DeviceControlRequest<AnyIoCtlCode>` in `ServerDriveIoRequest`.

### CODE REVIEW

File: crates/ironrdp-rdpdr/src/pdu/efs.rs [hunk 1]
```diff
@@ -2391,3 +2440,433 @@ impl ClientDriveQueryDirectoryResponse {
...
+    pub device_io_request: DeviceIoRequest,
+    pub fs_info_class_lvl: FileSystemInformationClassLevel,
+}
+
+impl ServerDriveQueryVolumeInformationRequest {
+    const NAME: &str = "DR_DRIVE_QUERY_VOLUME_INFORMATION_REQ";
+    const FIXED_PART_SIZE: usize = 4 /* FsInformationClass */ + 4 /* Length */ + 24 /* Padding */;
+
+    pub fn decode(dev_io_req: DeviceIoRequest, src: &mut ReadCursor<'_>) -> PduResult<Self> {
+        ensure_size!(in: src, size: Self::FIXED_PART_SIZE);
```
Comment (CBenoit) [inline crates/ironrdp-rdpdr/src/pdu/efs.rs:2462]: Nitpick: we have `ensure_fixed_part_size!(in: src)`, specifically for this. I’ll not block the PR for that, feel free to send a follow up PR if you prefer.

**Review by CBenoit — COMMENTED** [review]
You need to rebase on top of master before we can merge (currently the PR includes commits from another branch already merged)

**Review by CBenoit — APPROVED** [review]
Approving for auto-merge once PR is rebased and conflicts are resolved.

File: crates/ironrdp-rdpdr/src/pdu/efs.rs (hunk 1 above)
Comment (ibeckermayer) [reply crates/ironrdp-rdpdr/src/pdu/efs.rs:2462]: I just noticed that, and patched all those up in https://github.com/Devolutions/IronRDP/pull/298

**Comment by ibeckermayer** [comment]
> You need to rebase on top of master before we can merge (currently the PR includes commits from another branch already merged)

Oof, not quite sure what happened, I will make things make more sense.

**Pull request merged**